apps/api/src/controllers/applicationController.ts

```diff
@@ -252,3 +252,73 @@ export const submitRevision = async ({ applicationId }: { applicationId: number
 		return failure(message, error);
 	}
 };
+
+export const closeApplication = async ({
+	applicationId,
+	requesterId,
+	isDacMember = false,
+}: {
+	applicationId: number;
+	requesterId: string;
+	isDacMember?: boolean;
+}) => {
+	try {
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+		const appStateManager = new ApplicationStateManager(application);
+
+		// Check if application is already closed
+		if (appStateManager.state === ApplicationStates.CLOSED) {
+			return failure('Application is already closed.', 'StateConflict');
+		}
+
+		// Authorization logic based on state
+		const isApplicant = application.user_id === requesterId;
+		let closeResult;
+
+		switch (appStateManager.state) {
+			case ApplicationStates.DRAFT:
+				if (!isApplicant) {
+					return failure('Only the applicant can close the application in this state.', 'Unauthorized');
+				}
+				closeResult = await appStateManager.closeDraft();
+				break;
+			case ApplicationStates.INSTITUTIONAL_REP_REVIEW:
+				if (!isApplicant) {
+					return failure('Only the applicant can close the application in this state.', 'Unauthorized');
+				}
+				closeResult = await appStateManager.closeRepReview();
+				break;
+			case ApplicationStates.DAC_REVIEW:
+				if (!isApplicant && !isDacMember) {
+					return failure('Only the applicant or a DAC member can close the application in DAC Review.', 'Unauthorized');
+				}
+				closeResult = await appStateManager.closeDacReview();
+				break;
+			default:
+				return failure(`Cannot close application in state ${appStateManager.state}.`, 'InvalidState');
+		}
+
+		if (!closeResult.success) {
+			return failure(closeResult.message || 'Failed to close application.', 'StateTransitionError');
+		}
+
+		// Update the database with the new CLOSED state
+		const update = { state: ApplicationStates.CLOSED };
+		const updatedResult = await service.findOneAndUpdate({ id: applicationId, update });
```

Comment:
These are not needed, `appStateManager.closeDraft();` and other close functions already include an application update
See: https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/src/controllers/stateManager.ts#L119